src/python/nxt_python_asgi.c

```diff
@@ -493,14 +494,35 @@ nxt_py_asgi_request_handler(nxt_unit_request_info_t *req)
     }
 
     req->data = asgi;
+    ctx_data = req->ctx->data;
     target = &nxt_py_targets->target[req->request->app_target];
+    lifespan = ctx_data->target_lifespans[req->request->app_target];
+    state = PyObject_GetAttr(lifespan, nxt_py_state_str);
+    if (nxt_slow_path(state == NULL)) {
+        nxt_unit_req_alert(req, "Python failed to get 'state' attribute");
+        nxt_unit_request_done(req, NXT_UNIT_ERROR);
 
+        goto release_send;
+    }
+    newstate = PyDict_Copy(state);
```

Comment:
While there are too many blank lines already in the code base for my liking, I think one is deserved here after the if () {} block.